service/resource/endpoint/create.go

```diff
@@ -1,11 +1,133 @@
 package endpoint
 
-import "context"
+import (
+	"context"
+
+	"github.com/giantswarm/microerror"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	apiv1 "k8s.io/client-go/pkg/api/v1"
+)
 
 func (r *Resource) GetCreateState(ctx context.Context, obj, currentState, desiredState interface{}) (interface{}, error) {
-	return nil, nil
+	currentEndpoints, err := toEndpointSlice(currentState)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	desiredEndpoints, err := toEndpointSlice(desiredState)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	var createState = []Endpoint{}
+	for _, desiredEndpoint := range desiredEndpoints {
+		if !containsEndpoint(currentEndpoints, desiredEndpoint) {
+			createState = append(createState, desiredEndpoint)
+		}
+	}
+
+	if len(createState) == 0 {
+		return nil, nil
+	}
+
+	return createState, nil
 }
 
 func (r *Resource) ProcessCreateState(ctx context.Context, obj, createState interface{}) error {
+	endpointsToCreate, err := toEndpointSlice(createState)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+	for _, endpoint := range endpointsToCreate {
```

Comment:
This is largely taken from `k8s-endpoint-updater` and adapted to work in this operator.